### PR TITLE
Update slick

### PR DESCRIPTION
**Pull request by moradology**
This PR supersedes PR #1385 and adds the following changes:
1. local test ability is a must, so a script has been added which uses docker to launch a postgres instance which is sufficient to the current batch of tests in package `slick`
2. some minor formatting was changed in the slick config for tests
3. tminglei's postgres addons are now explicitly depended upon to the effect that copying and pasting portions of his codebase are no longer necessary
4. these changes have been merged with master up to this point so that they might be automatically merged into master

@adamkozuch's changes get my blessing and a +1, a second pair of eyes might be useful for the changes I've made. @lossyrob @echeipesh

### CODE REVIEW

File: slick/src/main/scala/geotrellis/slick/PostGisProjectionSupport.scala [hunk 1]
```diff
@@ -28,66 +28,71 @@
...
   type LINESTRING         = Projected[Line]
   type POLYGON            = Projected[Polygon]
   type GEOMETRYCOLLECTION = Projected[GeometryCollection]
 
-  implicit val geometryTypeMapper           = new ProjectedGeometryJdbcType[GEOMETRY]
-  implicit val pointTypeMapper              = new ProjectedGeometryJdbcType[POINT]
-  implicit val lineTypeMapper               = new ProjectedGeometryJdbcType[LINESTRING]
-  implicit val polygonTypeMapper            = new ProjectedGeometryJdbcType[POLYGON]
+  trait PostGISProjectionAssistants extends BasePostGISAssistants[GEOMETRY, POINT, LINESTRING, POLYGON, GEOMETRYCOLLECTION]
+trait PostGISProjectionImplicits {
```
Comment (echeipesh) [inline slick/src/main/scala/geotrellis/slick/PostGisProjectionSupport.scala]: indentation is off, threw me for a sec.

**Comment by echeipesh** [comment]
Really cool PR. I like the docker scripts a lot. Probably need a note at the top of the tests to mention that it needs to be run.

File: slick/src/test/scala/geotrellis/slick/PostgisSpec.scala [hunk 2]
```diff
@@ -52,147 +57,139 @@ class PostgisSpec extends FlatSpec with ShouldMatchers with TestDatabase {
   }
   val CityTable = TableQuery[City]
 
+  def createSchema() = db.run(CityTable.schema.create).futureValue
```
Comment (echeipesh) [inline slick/src/test/scala/geotrellis/slick/PostgisSpec.scala]: Maybe a try/catch here for connection error and augment it with message to use `scripts/slickTestDB` for local tests.

**Comment by lossyrob** [comment]
Superseded by #1496

**Pull request merged**